src/EFCore.PG/Query/Internal/NpgsqlQuerySqlGenerator.cs

```diff
@@ -417,8 +417,9 @@ public virtual Expression VisitPgFunction([NotNull] PgFunctionExpression sqlFunc
                     .Append(".");
             }
 
+            // TODO: Quote user-defined function names with upper-case
             Sql
-                .Append(_sqlGenerationHelper.DelimitIdentifier(sqlFunctionExpression.FunctionName))
+                .Append(sqlFunctionExpression.FunctionName)
```

Comment:
What's blocking here?

Reply:
There's a painful subject of built-in vs. user functions, functions in schema and quoting. If you take a look at SqlFunctionExpression upstream, you'll see that functions without a schema are assumed to be "built-in", which means that they don't get quoted. This largely comes from SQL Server (where you're not allowed to quote built-in functions), but it also kinda make sense for us: we'd like to be able to generate function names with uppercase letters -  `MAX()` or `ST_Intersect()` - but if we quote PostgreSQL will fail to find them.

Basically PgFunctionExpression needs to have more or less the same logic as SqlFunctionExpression does with regards to all this, but I didn't want to block on it. We're also only using PgFunctionExpression in one spot, so this isn't very important.

Reply:
Makes sense, thanks for the explanation!